packages/python-packages/apiview-copilot/src/mention/_open_parser_issue_workflow.py

```diff
@@ -0,0 +1,84 @@
+import json
+import prompty
+from src._github_manager import GithubManager
+from src._utils import get_prompt_path
+from ._base import MentionWorkflow
+
+
+class OpenParserIssueWorkflow(MentionWorkflow):
+    prompty_filename = "parse_conversation_to_github_issue.prompty"
+    summarize_prompt_file = "summarize_github_actions.prompty"
+    deduplication_prompt_file = "deduplicate_parser_issue.prompty"
+    
+    REPO_OWNER = "tjprescott"
+    REPO_NAME = "azure-sdk-tools"
+
+    def execute_plan(self, plan: dict):
+        """Execute the parser issue workflow"""
+        recent_issues = self._fetch_recent_parser_issues()
+        dedup_result = self._check_for_duplicate_issue(plan, recent_issues)
+        
+        if dedup_result["action"] == "no-op":
+            return {
+                "action": "no-op",
+                "existing_issue": dedup_result["existing"],
+            }
+        
+        issue = self._create_parser_issue(plan)
+        return {
+            "action": "created",
+            "issue": issue
+        }
+
+    def _fetch_recent_parser_issues(self):
+        """Fetch recent open parser issues from GitHub."""
+        client = GithubManager.get_instance()
+        issues = client.search_issues(
+            owner=self.REPO_OWNER,
+            repo=self.REPO_NAME,
+            query="label:parser is:issue state:open"
+        )
+        return issues
+
+    def _check_for_duplicate_issue(self, plan: dict, recent_issues: list) -> dict:
+        """Check if the issue already exists with deduplication prompt."""
+        dedup_prompt_path = get_prompt_path(folder="mention", filename=self.deduplication_prompt_file)
+        error_context = f"{plan.get('title')}\n\n{plan.get('body')}"
+        dedup_inputs = {
+            "language": self.language,
+            "package_name": self.package_name,
+            "code": self.code,
+            "error_context": error_context,
+            "existing_issues": self._format_issues_for_dedup(recent_issues)
+        }
+        raw_dedup = prompty.execute(dedup_prompt_path, inputs=dedup_inputs)
+        
+        try:
+            return json.loads(raw_dedup)
+        except json.JSONDecodeError:
```

Comment:
If the output is in a bad format, you should error out. You can also force prompty to use a json schema so it can't return bad JSON.